ui/app/AppLayouts/Wallet/views/AdvancedAddAccountView.qml

```diff
@@ -0,0 +1,354 @@
+    property string mnemonicText: getSeedPhraseString()
+    property string errorString: ""
+    property bool isValid: addAccountType === AdvancedAddAccountView.AddAccountType.ImportSeedPhrase ? grid.isValid :
+                           addAccountType === AdvancedAddAccountView.AddAccountType.ImportPrivateKey ? (privateKey.text !== "" && privateKey.valid) : true
+
+    enum AddAccountType {
+        GenerateNew,
+        ImportSeedPhrase,
+        ImportPrivateKey
+    }
+
+    function reset() {
+        mnemonicText = ""
+        errorString = ""
+        select.currentIndex = 0
+        addAccountType = AdvancedAddAccountView.AddAccountType.GenerateNew
+        privateKey.text = ""
+        privateKey.reset()
+        for(var i = 0; i < grid.model; i++) {
+            if(grid.itemAtIndex(i)) {
+                grid.itemAtIndex(i).textEdit.text =  ""
+                grid.itemAtIndex(i).textEdit.reset()
+            }
+        }
+    }
+
+    function validate() {
+        errorString = "";
+        if(addAccountType == AdvancedAddAccountView.AddAccountType.ImportSeedPhrase) {
+            mnemonicText = getSeedPhraseString()
+
+            if (!Utils.isMnemonic(mnemonicText)) {
+                //% "Invalid seed phrase"
+                errorString = qsTrId("custom-seed-phrase")
+            } else {
+                errorString = onboardingModule.validateMnemonic(mnemonicText)
+                const regex = new RegExp('word [a-z]+ not found in the dictionary', 'i');
+                if (regex.test(errorString)) {
+                    //% "Invalid seed phrase"
+                    errorString = qsTrId("custom-seed-phrase") + '. ' +
+                            //% "This seed phrase doesn't match our supported dictionary. Check for misspelled words."
+                            qsTrId("custom-seed-phrase-text-1")
+                }
+            }
+            return errorString === ""
+        }
+        else if(addAccountType == AdvancedAddAccountView.AddAccountType.ImportPrivateKey) {
+            if (privateKey.text === "") {
+                //% "You need to enter a private key"
+                errorString = qsTrId("you-need-to-enter-a-private-key")
+            } else if (!Utils.isPrivateKey(privateKey.text)) {
+                //% "Enter a valid private key (64 characters hexadecimal string)"
+                errorString = qsTrId("enter-a-valid-private-key-(64-characters-hexadecimal-string)")
+            } else {
+                errorString = ""
+            }
+            return errorString === ""
+        }
+        return true
+    }
+
+    function getSeedPhraseString() {
+        var seedPhrase = ""
+        for(var i = 0; i < grid.model; i++) {
+            seedPhrase += grid.itemAtIndex(i).text + " "
+        }
+        return seedPhrase
+    }
+
+    QtObject {
+        id: _internal
+        property int seedPhraseInputHeight: 44
+        property int seedPhraseInputWidth: 220
```

Comment:
Very good idea to keep values in the object. I think you can move more values here, eg. radius, pixel size ...